README.md

```diff
@@ -89,6 +89,42 @@ input = ["KEY_F8"]
 output = ["KEY_MUTE"]
 ```
 
+`REL_*` and `ABS_*` input types are also supported. These have an associated 
+positive or negative value depending on the relative direction of the input.
+You can optionally specify the relative direction by appending a "+" or "-" 
+to the input name. For example, if you wanted to remap your scroll wheel to 
+a keypress:
+
+```toml
+[[remap]]
+input = ["REL_WHEEL+"]
+output = ["KEY_UP"]
+
+[[remap]]
+input = ["REL_WHEEL-"]
+output = ["KEY_DOWN"]
+```
+
+Mapping relative inputs to relative inputs is also possible, you can include 
+a scaling factor to both the input value and output. The input value is divided 
+by the input scaling factor, and the output is multiplied by the output scaling 
+factor. Specify a negative scale factor on the output to invert the input. If you 
+provide an input scale, you need to remap both relative directions separately:
+
+```toml
+[[remap]]
+input = ["REL_WHEEL"]
+output = ["REL_X-2"]
+
+[[remap]]
+input = ["REL_HWHEEL_HI_RES+4"]
+output = ["REL_Y+2"]
```

Comment:
I'm having a hard time grokking what this PR is doing overall, by which I mean: I get the gist of what the higher level goal is, but when it comes to the implementation, the details are not at all clear to me.

I think it is due in part to overloading what was a simple keycode type with something that is conceptually richer, but which in the current implementation feels a little bit like it was "finagled in".

What I think might help is to reduce the scope of the changes around the `KeyCodeWrapper` and to make it explicitly handle the situations that are necessary.

For example, rather than a struct it could be an enum, which would allow tracking the scale only for the relative case:

```rust
#[derive(PartialEq, Eq, Hash, Clone, Debug)]
enum EventWrapper {
   Key(KeyCode),
   Relative {
      code: KeyCode,
      scale: NotNan<f32>,
   },
   Absolute(KeyCode),
}
```

This sort of change should make it a bit clearer in the consuming code what's going on.